src/core/public/http/http_setup.ts

```diff
@@ -118,6 +158,14 @@ export const setup = (
     return body;
   }
 
+  async function fetch(path: string, options: HttpFetchOptions = {}) {
+    try {
+      return await interceptResponse(fetcher(path, options));
+    } catch (err) {
+      return interceptResponse(Promise.resolve(await interceptRejection(err)));
```

Comment:
```suggestion
      return interceptResponse(interceptRejection(err));
```

Reply:
This is being awaited here in order to force a throw **prior* to falling into `interceptResponse`. I'll add a comment.